models/ed/R/model2netcdf.ED2.R

```diff
@@ -940,9 +966,23 @@ put_E_values <- function(yr, nc_var, out, lat, lon, begins, ends, pft_names, ...
   
   # ----- fill list
   
+  # setup output time and time bounds
+  sub_dates <- seq(lubridate::floor_date(begins,"month"), by = "month", length.out = dim(out[[1]])[1] )
+  days_per_month <- as.vector(lubridate::days_in_month(sub_dates))
+  if (lubridate::yday(begins)!=lubridate::yday(sub_dates[1])) {
+    temp <- days_per_month[1] - ((lubridate::yday(begins)-lubridate::yday(sub_dates[1])))
+    days_per_month[1] = temp
+    sub_dates[1]=begins
+  }
+  jdates <- lubridate::yday(sub_dates)
+  dtvals <- jdates-1 # convert to 0 index
+  bounds <- array(data=NA, dim=c(length(dtvals),2))
+  bounds[,1] <- dtvals; bounds[,2] <- bounds[,1]+days_per_month; bounds <- round(bounds,4)  # create time bounds for each timestep in t, t+1; t+1, t+2... format
+  #head(bounds); tail(bounds)
+  #
```

Comment:
I'd need to look more carefully, but from a first pass, this block seems very similar to an earlier one. Is there any way this could be refactored to avoid code duplication?

Reply:
Not sure that is possible with this one.  The way @istfer created the two sub functions for -T- and -E- outputs, its hard to pass info - but is possible.  Not only that, these bounds are different from the previous as these represent the start/end dates of the monthly output not daily, as such these bound values are not the same as previous.

Reply:
Fair enough. Might be useful to add a small comment highlighting differences between the two, so someone like me doesn't come along with a refactoring wrecking ball in two years and accidentally changes an important part of the logic.

Reply:
Will do....Probably a caution statement that informs a future editor that they are not the same. Plus I will change the second to be month_bounds instead of bounds